.github/workflows/ci-publish.yml

```diff
@@ -0,0 +1,127 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: CI Release artifacts publish
+
+on:
+  push:
+    tags:
+      - v*.*.*
+
+jobs:
+  build-all:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'zulu'
+          java-version: '8.0.312+7'
+          java-package: jdk
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+          architecture: x64
+      - name: Install pip3
+        run: sudo apt-get update && sudo apt-get install -y python3-pip
+      - name: Set JanusGraph versions environment variables
+        run: |
+          export RELEASE_TAG=${{github.ref_name}}
+          export JG_VER="${RELEASE_TAG//v/}"
+          export JG_DIST_NAME="janusgraph-${JG_VER}"
+          export JG_FULL_DIST_NAME="janusgraph-full-${JG_VER}"
+          export JG_DOC_DIST_NAME="${JG_DIST_NAME}-doc"
+          echo "RELEASE_TAG=${RELEASE_TAG}" >> $GITHUB_ENV
+          echo "JG_VER=${JG_VER}" >> $GITHUB_ENV
+          echo "JG_DIST_NAME=${JG_DIST_NAME}" >> $GITHUB_ENV
+          echo "JG_FULL_DIST_NAME=${JG_FULL_DIST_NAME}" >> $GITHUB_ENV
+          echo "JG_DOC_DIST_NAME=${JG_DOC_DIST_NAME}" >> $GITHUB_ENV
+      - name: Configure GPG Key
+        run: |
+          echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
+        env:
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+      - name: Set OSSR credentials
+        run: |
+          mkdir -p ~/.m2/
+          echo "<settingsSecurity><master>$MASTER_PASSWORD</master></settingsSecurity>" > ~/.m2/settings-security.xml
+          echo "<settings><servers><server><id>ossrh</id><username>$OSSRH_USERNAME</username><password>$OSSRH_PASSWORD</password></server></servers></settings>" > ~/.m2/settings.xml
+        env:
+          MASTER_PASSWORD: ${{ secrets.MASTER_PASSWORD }}
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+      # TODO: the step "Make JanusGraph JAR artifacts" should also create signature key for full JanusGraph bundle but for some
+      # reason it doesn't sign full distribution (only standard distribution).
+      # After this is fixed we can delete the step "Sign JanusGraph full distribution artifact".
+      - name: Make JanusGraph JAR artifacts
+        run: mvn clean install -Pjanusgraph-release -DskipTests=true
```

Comment:
Wouldn't great to build also java 11 variant?

Reply:
We are not publishing java-11 builds yet, so I didn't include it yet but that's definitely possible. Do you think we need to add java-11 builds to 0.6.1 release?
Also, we are not publishing java-11 artifacts to Sonatype because they will overwrite java-8 artifacts. So, it should probably be another job which lunches with java-11, builds artifacts but doesn't publish them to sonatype. Only attaches the release bundle to GitHub release. That said, I didn't want to complicate things with java-11 at this moment and thought we could do it in another PR after some discussion